src/app/index.js

```diff
@@ -90,8 +90,43 @@ export default class App extends Widget {
         this.dom.loadingOverlay.classList.remove('error', 'done');
 
         return fetch(explicitData ? 'data:application/json,{}' : url)
-            .then(res => {
-                return explicitData || res.json();
+            .then(async res => {
+                if (explicitData) {
+                    return explicitData;
+                }
+
+                const contentLength = Number(res.headers.get('content-length'));
+
+                if (contentLength) {
+                    const reader = res.body.getReader();
+                    const chunks = [];
+                    let recievedLength = 0;
+
+                    while (true) {
+                        const { done, value } = await reader.read();
+
+                        if (done) {
+                            break;
+                        }
+
+                        chunks.push(value);
+                        recievedLength += value.length;
+
+                        this.dom.loadingOverlay.innerHTML = `Loading data (${Math.floor((recievedLength / contentLength) * 100)}%)...`;
+                    }
+
+                    const chunksArr = new Uint8Array(recievedLength);
+                    let pos = 0;
+
+                    for (let chunk of chunks) {
```

Comment:
Not sure, but let's give it a chance to release some memory:
```
while (chunks.length) {
    const chunk = cuhnks.pop();
    chunksArr.set(chunk, pos);
    pos += chunk.length;
}

const decoded = new TextDecoder('utf-8').decode(chunksArr);

chunksArr = null;

return JSON.parse(decoded);
```

Reply:
OK, let's try it this way